In [5]:
dic1={'a':5,'b':4,'c':9,'d':3,'e':8}
dic1

{'a': 5, 'b': 4, 'c': 9, 'd': 3, 'e': 8}

In [6]:
dic1.values()

dict_values([5, 4, 9, 3, 8])

In [17]:
dic1={'a':5,'b':4,'c':9,'d':3,'e':8}
for i in dic1.values():
    print(i)

5
4
9
3
8


# 2. Write a Python script to add a key to a dictionary. 
#Sample Dictionary : {0: 10, 1: 20}
#Expected Result : {0: 10, 1: 20, 2: 30}

In [16]:
sample_dic={0:10,1:20}
print("sample dic: ",sample_dic)
sample_dic[2]=30
print("expected result: ",sample_dic)

sample dic:  {0: 10, 1: 20}
expected result:  {0: 10, 1: 20, 2: 30}


# 3. Write a Python script to concatenate following dictionaries to create a new one. 
Sample Dictionary :
dic1={1:10, 2:20}
dic2={3:30, 4:40}
dic3={5:50,6:60}
Expected Result : {1: 10, 2: 20, 3: 30, 4: 40, 5: 50, 6: 60}


In [32]:
ex_dic={}
dic1={1:10,2:20}
dic2={3:30,4:40}
dic3={5:50,6:60}
ex_dic.update(dic1)
ex_dic.update(dic2)
ex_dic.update(dic3)
print(ex_dic)

{1: 10, 2: 20, 3: 30, 4: 40, 5: 50, 6: 60}


# list ##

In [33]:
# 1.Write a Python program to sum all the items in a list.

In [60]:
list=[1,2,15,7,10]
sum=0
for i in list:
    sum=sum+i
print("sum is: ",sum)

sum is:  35


In [61]:
# 2.Write a Python program to multiply all the items in a list.

In [62]:
multi=1
for i in list:
   multi=multi*i
print("multiply is: ",multi)


multiply is:  2100


In [64]:
# 3.Write a Python program to get the largest number from list

large=max(list)
print("largest number is :",large)

    

largest number is : 15


In [65]:
### 4. Write a Python program to get the smallest number from a list.
small=min(list)
print("smallest number is: ",small)

smallest number is:  1


In [66]:
## 5.Write a Python program to count the number of strings where the string length is 2 or more and the first and last character are same from a given list of strings. 
#Sample List : ['abc', 'xyz', 'aba', '1221']
#Expected Result : 2


In [118]:
list=['abc','xyx','aba','1221']
count=0
for i in list:
    if len(i)>=2 and i[0]==i[-1]:
        count+=1
print(count)

    

3


In [ ]:
##6.Write a Python program to get a list, sorted in increasing order by the last element in each tuple from a given list of non-empty tuples. 
#Sample List : [(2, 5), (1, 2), (4, 4), (2, 3), (2, 1)]
#Expected Result : [(2, 1), (1, 2), (2, 3), (4, 4), (2, 5)]


In [190]:
sample_list = [(2, 5), (1, 2), (4, 4), (2, 3), (2, 1)]

sorted_list=[(x[0],x[1]) for x in sample_list]                                 ####error

print("List of tuples with sorted second elements:", sorted_list)

List of tuples with sorted second elements: [(2, 5), (1, 2), (4, 4), (2, 3), (2, 1)]


In [191]:
# 7.Write a Python program to remove duplicates from a list. 


In [213]:
sample_list=[1,2,3,4,1,5,5,6,8,2]
print("original list: ",sample_list)
#new=list(set(sample_list))
new=[]
for i in sample_list:
   if i not in new:
      new.append(i)
print("new list: ",new)

original list:  [1, 2, 3, 4, 1, 5, 5, 6, 8, 2]
new list:  [1, 2, 3, 4, 5, 6, 8]


In [214]:
##8.  Write a Python program to check a list is empty or not. 


In [215]:
if len(sample_list)==0:
    print("empty..list")
else:
    print("non empty...")

non empty...


In [221]:
sample_list=[1,2,3,4,1,5,5,6,8,2]
#sample_list=[]
def result(sample_list):
    if not sample_list:
        return True
    else:
        return False
print("result....",result(sample_list))

result.... False


#9. Write a Python program to clone or copy a list. 


In [222]:
new=sample_list.copy()
print(new)


[1, 2, 3, 4, 1, 5, 5, 6, 8, 2]


In [223]:
new=sample_list[:]
new

[1, 2, 3, 4, 1, 5, 5, 6, 8, 2]

# 10. Write a Python program to find the list of words that are longer than n from a given list of words. 


In [259]:
list=['banana','mango','chiku','kiwi','watermelon','pinapple']
n=int(input("enter number :"))
long_list=[]
for i in list:
    if len(i)>=n:
        long_list.append(i)
        #print(long_list)
print(long_list)

enter number :6
['banana', 'watermelon', 'pinapple']


11. Write a Python function that takes two lists and returns True if they have at least one common member. 


In [306]:
list1=['ram','shayam','kishan','ghanshyam']
list2=['ena','mina','dika','ram','shyam']
for i in list1:
    if i in list2:
        print("True")
        break
    else:
        print("false")
        break

True


12. Write a Python program to print a specified list after removing the 0th, 4th and 5th elements. 
Sample List : ['Red', 'Green', 'White', 'Black', 'Pink', 'Yellow']
Expected Output : ['Green', 'White', 'Black']


In [2]:
sample_list=['red','green','white','black','pink','yellow']
index_remove=[0,4,5]
new_list=[]
for i in range(len(sample_list)):
    if i not in index_remove:
        new_list.append(sample_list[i])
print(new_list)


['green', 'white', 'black']


### 13. Write a Python program to print the numbers of a specified list after removing even numbers from it.

In [1]:
sample_list=[1,10,5,4,6,9,15,50,13,14]
new_list=[]
for i in sample_list:
    #print(i)
    if i%2!=0:
        new_list.append(i)
print(new_list)


[1, 5, 9, 15, 13]


14. Write a Python program to generate and print a list of first and last 5 elements where the values are square of numbers between 1 and 30 (both included). 


In [24]:
new_list=[]
for i in range(1,31):
    i=i**2
    new_list.append(i)
first_five=new_list[:5]
last_five=new_list[-5:]
print(first_five+last_five)
    

[1, 4, 9, 16, 25, 676, 729, 784, 841, 900]


15. Write a Python program to generate and print a list except for the first 5 elements, where the values are square of numbers between 1 and 30 (both included). 


In [25]:
new_list=[]
for i in range(1,31):
    i=i**2
    new_list.append(i)
print(new_list[5:])

[36, 49, 64, 81, 100, 121, 144, 169, 196, 225, 256, 289, 324, 361, 400, 441, 484, 529, 576, 625, 676, 729, 784, 841, 900]


16. Write a Python program to get the difference between the two lists. 


In [38]:
list1=[1,5,10,3]
list2=[5,4,10]

diff=list(set(list1)-set(list2))
diff

[1, 3]

#17. Write a Python program access the index of a list. 

In [51]:

list=["ram",'shyam','kishan']
for i in range(len(list)):
    print("list[{}] ".format(i),list[i])

list[0]  ram
list[1]  shyam
list[2]  kishan


18. Write a Python program to convert a list of characters into a string.

In [60]:
list=['a','b','c']
string=list[0]+list[1]+list[2]
print(string)

abc


19. Write a Python program to append a list to the second list. 


In [61]:
list1=['one','two','three']
list2=[1,2,3]
list2.append(list1)
print(list2)

[1, 2, 3, ['one', 'two', 'three']]


In [66]:
list1=['one','two','three']
list2=[1,2,3]
for i in range(len(list1)):
    list2.append(list1[i])
print(list2)

[1, 2, 3, 'one', 'two', 'three']


20. Write a Python program to find the second smallest number in a list. 



In [15]:
n=[5,24,9,13,2,7,10,6]
n.sort()
print("second samllest number: ",n[1])

second samllest number:  5


21.Write a Python program to find the second largest number in a list. 


In [18]:
my_list = [5, 2, 8, 1, 9, 3]
my_list.sort()
print("Second largest number:", my_list[-2])

Second largest number: 8


22. Write a Python program to count the number of elements in a list within a specified range. 


In [52]:
list=['mira','ram','kishan','shyam','mohan']
lower=1
upper=4
result=[]
for i in range(len(list)):
  
    if i>lower and i<upper:
        result.append(list[i])
print("range of [{}:{}] : {} elements".format(lower,upper,len(result)))



range of [1:4] : 2 elements


23. Write a Python program to find common items from two lists. 


In [57]:
list1=[1,5,8,15,17,26,64,7,9]
list2=[5,6,2,10,5,25,64,4,15]
for i in list1:
    if i in list2:
        print(i)

5
15
64


24.Write a Python program to change the position of every n-th value with the (n+1)th in a list. 
Sample list: [0,1,2,3,4,5]
Expected Output: [1, 0, 3, 2, 5, 4]


In [96]:
slist=[0,1,2,3,4,5]
for n in range(0,len(slist),2):
    slist[n],slist[n+1]=slist[n+1],slist[n]
print(slist)
    

[1, 0, 3, 2, 5, 4]


25. Write a Python program to convert a list of multiple integers into a single integer. 
Sample list: [11, 33, 50]
Expected Output: 113350


In [109]:
slist=[11,30,50]
str(slist[0])+str(slist[1])+str(slist[2])

'113050'

26. Write a Python program to split a list based on first character of word. 


In [252]:
word_list = ["apple", "banana", "ball", "cat", "dog", "elephant"]
word_dic=[]
for i in word_list:
   # print(i)
    first_char=i[0]
   # print(first_char)
    word_dic=i
    print("first character {} of element:  {} ".format(first_char,word_dic))
        

first character a of element:  apple 
first character b of element:  banana 
first character b of element:  ball 
first character c of element:  cat 
first character d of element:  dog 
first character e of element:  elephant 


27.Write a Python program to select the odd items of a list.


In [270]:
list=[1,2,3,14,5,6,17,8]
odd_item=[]
for i in list:
    if i%2!=0:
        odd_item.append(i)
print("odd iem: ",odd_item)

odd iem:  [1, 3, 5, 17]


28. Write a Python program to print a nested lists (each list on a new line) using the print() function. 


In [273]:
list=[[1,2,3],[4,5,6],['seven',8,'nine']]
for i in list:
    print(i)

[1, 2, 3]
[4, 5, 6]
['seven', 8, 'nine']


29. Write a Python program to insert an element before each element of a list. 


In [280]:
list=[1,2,3,4,5,6]
element=str(input("enter element to insert: "))
new_list=[]
for i in list:
    new_list.append(element)
    new_list.append(i)
print(new_list)

enter element to insert: h
['h', 1, 'h', 2, 'h', 3, 'h', 4, 'h', 5, 'h', 6]


30. Write a Python program to split a list every Nth element. 
Sample list: ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n']
Expected Output: [['a', 'd', 'g', 'j', 'm'], ['b', 'e', 'h', 'k', 'n'], ['c', 'f', 'i', 'l']]


In [337]:
slist=['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n']
nlist=[]
n=3
for i in range(n):
    #print(i)
    nlist.append(slist[i::3])
print(nlist)


[['a', 'd', 'g', 'j', 'm'], ['b', 'e', 'h', 'k', 'n'], ['c', 'f', 'i', 'l']]


31. Write a Python program to compute the difference between two lists. 
Sample data: ["red", "orange", "green", "blue", "white"], ["black", "yellow", "green", "blue"]
Expected Output:
Color1-Color2: ['white', 'orange', 'red']
Color2-Color1: ['black', 'yellow']


In [46]:
color1=["red", "orange", "green", "blue", "white"]
color2=["black", "yellow", "green", "blue"]
color=[]
diff1=set(color1)-set(color2)
color.append(diff1)
print(color)
diff2=set(color2)-set(color1)
print(diff2)

[{'red', 'orange', 'white'}]
{'black', 'yellow'}


32. Write a Python program to insert a given string at the beginning of all items in a list. 
Sample list : [1,2,3,4], string : emp
Expected output : ['emp1', 'emp2', 'emp3', 'emp4']


In [63]:
slist=[1,2,3,4]

string='emp'

new_list=[(string)+str(i) for i in slist]    
new_list

['emp1', 'emp2', 'emp3', 'emp4']

33. Write a Python program to find the items starts with specific character from a given list. 
Expected Output:
Original list:
['abcd', 'abc', 'bcd', 'bkie', 'cder', 'cdsw', 'sdfsd', 'dagfa', 'acjd']
Items start with a from the said list:
['abcd', 'abc', 'acjd']
Items start with d from the said list:
['dagfa']
Items start with w from the said list:
[]

In [79]:
original_list = ['abcd', 'abc', 'bcd', 'bkie', 'cder', 'cdsw', 'sdfsd', 'dagfa', 'acjd']
a=[i for i in original_list if i.startswith('a')]    
print("start with 'a':" ,a)    
d=[i for i in original_list if i.startswith('d')]    
print("start with 'a':" ,d)     
w=[i for i in original_list if i.startswith('w')]    
print("start with 'a':" ,w)    

start with 'a': ['abcd', 'abc', 'acjd']
start with 'a': ['dagfa']
start with 'a': []


34. Write a Python program to insert an element at a specified position into a given list. 
Original list:
[1, 1, 2, 3, 4, 4, 5, 1]
After inserting an element at kth position in the said list:
[1, 1, 12, 2, 3, 4, 4, 5, 1]

In [81]:
list1=[1, 1, 2, 3, 4, 4, 5, 1]
element=input("enter element to insert: ")
position=int(input("enter position: "))
list1.insert(position,element)
print(list1)

enter element to insert: 50
enter position: 2
[1, 1, '50', 2, 3, 4, 4, 5, 1]


35. Write a Python program to read a square matrix from console and print the sum of matrix primary diagonal. Accept the size of the square matrix and elements for each column separated with a space (for every row) as input from the user. 
Input the size of the matrix: 3
2 3 4
4 5 6
3 4 7
Sum of matrix primary diagonal:
14

In [102]:
row=int(input("enter row: "))
colum=int(input("enter colum"))
matrix=[]
for i in range(row):
    c_row=[]
    for j in range(colum):
        n=int(input("enter element: "))
        c_row.append(n)
    matrix.append(c_row)
for array in matrix:
    print(array)

primary_diagonal_sum = sum(matrix[i][i] for i in range(min(row, colum)))

print("Sum of matrix primary diagonal:")
print(primary_diagonal_sum)



enter row: 2
enter colum2
enter element: 3
enter element: 5
[3, 5]
enter element: 4
enter element: 8
[4, 8]
[3, 5]
[4, 8]
Sum of matrix primary diagonal:
11


36. Write a Python program to Zip two given lists of lists. 
Original lists:
[[1, 3], [5, 7], [9, 11]]
[[2, 4], [6, 8], [10, 12, 14]]
Zipped list:
[[1, 3, 2, 4], [5, 7, 6, 8], [9, 11, 10, 12, 14]]

In [109]:
list1=[[1, 3], [5, 7], [9, 11]]
list2=[[2, 4], [6, 8], [10, 12, 14]]
new_list=[[a+b] for a,b in zip(list1,list2)]

new_list

[[[1, 3, 2, 4]], [[5, 7, 6, 8]], [[9, 11, 10, 12, 14]]]

37. Write a Python program to reverse strings in a given list of string values. 
Original lists:
['Red', 'Green', 'Blue', 'White', 'Black']
Reverse strings of the said given list:
['deR', 'neerG', 'eulB', 'etihW', 'kcalB']

In [125]:
list1=['Red', 'Green', 'Blue', 'White', 'Black']
rev_list=[s[::-1] for s in list1]
print(rev_list)

['deR', 'neerG', 'eulB', 'etihW', 'kcalB']


38. Write a Python program to remove specific words from a given list. 
Original list:
['red', 'green', 'blue', 'white', 'black', 'orange']
Remove words:
['white', 'orange']
After removing the specified words from the said list:
['red', 'green', 'blue', 'black']

In [126]:
list1=['red', 'green', 'blue', 'white', 'black', 'orange']
list1.remove('white')
list1.remove('orange')
list1

['red', 'green', 'blue', 'black']

39. Write a Python program to find the last occurrence of a specified item in a given list. 
Original list:
['s', 'd', 'f', 's', 'd', 'f', 's', 'f', 'k', 'o', 'p', 'i', 'w', 'e', 'k', 'c']
Last occurrence of f in the said list:
7
Last occurrence of c in the said list:
15
Last occurrence of k in the said list:
14
Last occurrence of w in the said list:
12

In [144]:
list1=['s', 'd', 'f', 's', 'd', 'f', 's', 'f', 'k', 'o', 'p', 'i', 'w', 'e', 'k', 'c']
f=len(list1)-1-list1[::-1].index('f')
c=len(list1)-1-list1[::-1].index('c')
index=list1[::-1].index('f')
print("index ",index)
print("last occurrence of 'f': ",f)
    
print("last occurrence of c: ",c)

index  8
last occurrence of 'f':  7
last occurrence of c:  15


# *Dictionary*

1. Write a Python script to sort (ascending and descending) a dictionary by value. 


In [27]:
my_dict = {1: 10, 2: 15, 3: 5, 4: 9, 5: 13, 6: 7}

def get_value(kv_pair):
    return kv_pair[1]

sorted_items = sorted(my_dict.items(), key=get_value)
sorted_dict_asc = {k: v for k, v in sorted_items}
print("Ascending order:", sorted_dict_asc)

sorted_items_desc = sorted(my_dict.items(), key=get_value, reverse=True)
sorted_dict_dsc = {k: v for k, v in sorted_items_desc}
print("Descending order:", sorted_dict_dsc)

Ascending order: {3: 5, 6: 7, 4: 9, 1: 10, 5: 13, 2: 15}
Descending order: {2: 15, 5: 13, 1: 10, 4: 9, 6: 7, 3: 5}


2. Write a Python script to add a key to a dictionary. 
Sample Dictionary : {0: 10, 1: 20}
Expected Result : {0: 10, 1: 20, 2: 30}


In [28]:
my_dict={0:10,1:20}
my_dict[2]=30

In [29]:
print(my_dict)

{0: 10, 1: 20, 2: 30}


3. Write a Python script to concatenate following dictionaries to create a new one. 
Sample Dictionary :
dic1={1:10, 2:20}
dic2={3:30, 4:40}
dic3={5:50,6:60}
Expected Result : {1: 10, 2: 20, 3: 30, 4: 40, 5: 50, 6: 60}


In [43]:
dic1={1:10,2:20}
dic2={3:30,4:40}
dic3={5:50,6:60}
new_dic={}
##new_dic=dic1,dic2,dic3
new_dic.update(dic1)
new_dic.update(dic2)
new_dic.update(dic3)
print(new_dic)


{1: 10, 2: 20, 3: 30, 4: 40, 5: 50, 6: 60}


4. Write a Python script to check whether a given key already exists in a dictionary.


In [61]:
dict1={1:'banana',2:'mango',3:'chiku',4:'kiwi'}
key=int(input("give key: "))
keys=dict1.keys()
if key in keys:
    print("already exist in dictionary")
else:
    print("not exist in dictionary")

give key: 3
already exist in dictionary


5.Write a Python script to generate and print a dictionary that contains a number (between 1 and n) in the form (x, x*x). 
Sample Dictionary ( n = 5) :
Expected Output : {1: 1, 2: 4, 3: 9, 4: 16, 5: 25}

In [77]:
n=int(input("enter number :"))
new_dic={}
for i in range(1,n+1):
    new_dic[i]=i*i
print(new_dic)

enter number :5
{1: 1, 2: 4, 3: 9, 4: 16, 5: 25}


6. Write a Python script to print a dictionary where the keys are numbers between 1 and 15 (both included) and the values are square of keys. 
Sample Dictionary
{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64, 9: 81, 10: 100, 11: 121, 12: 144, 13: 169, 14: 196, 15: 225}


In [79]:
new_dic={}
for i in range(1,15+1):
    new_dic[i]=i*i
print(new_dic)

{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64, 9: 81, 10: 100, 11: 121, 12: 144, 13: 169, 14: 196, 15: 225}


7. . Write a Python script to merge two Python dictionaries. 

In [82]:
dic1={1:'mango',2:'kiwi'}
dic2={3:'banana',4:'chiku'}
fruit={}
fruit.update(dic1)              ###update is use to add multi dictionary in one
fruit.update(dic2)
fruit

{1: 'mango', 2: 'kiwi', 3: 'banana', 4: 'chiku'}

8.  Write a Python program to sum all the items in a dictionary.


In [2]:
dict1={1:10,2:14,3:5,4:9,5:12}
print("sum of value: ",sum(dict1.values()))

sum of value:  50


In [3]:
sum=0
for i in dict1.values():
    sum=sum+i
print("sum is :",sum)

sum is : 50


9. Write a Python program to multiply all the items in a dictionary.

In [4]:
multiply=1
for i in dict1.values():
    multiply=multiply*i
print(multiply)

75600


10. Write a Python program to remove a key from a dictionary. 


In [15]:
dict1={1:10,2:14,3:5,4:9,5:12}

#dict1.pop(2)
del dict1[2]
print(dict1)

{1: 10, 3: 5, 4: 9, 5: 12}


11. Write a Python program to map two lists into a dictionary. 

In [23]:
key=[1,2,3]
value=['mango','banana','kiwi']
dict1=dict(zip(key,value))              ##### zip fubctin used to join 2 list
print(dict1)

{1: 'mango', 2: 'banana', 3: 'kiwi'}


12. Write a Python program to sort a given dictionary by key.

In [29]:
my_dict={2:"ram",5:"kishan",3:"mohan",9:"shyam"}
sorted_dict = {k: my_dict[k] for k in sorted(my_dict)}

print("Sorted Dictionary by key:", sorted_dict)



Sorted Dictionary by key: {2: 'ram', 3: 'mohan', 5: 'kishan', 9: 'shyam'}


13.  Write a Python program to get the maximum and minimum value in a dictionary. 


In [56]:
dict1={1:5,2:8,3:15,4:14,5:50}
print("max value: ",max(dict1.values()))
print("min value: ",min(dict1.values()))


max value:  50
min value:  5


14. Write a Python program to remove duplicates from Dictionary. 


In [2]:
my_dict = {'a': 1, 'b': 2, 'c': 1, 'c': 3, 'e': 2}
new_dict={}

for i,j in my_dict.items():
    if i not in new_dict:
        new_dict[i]=j

print("after removing duplicate: ",new_dict)

after removing duplicate:  {'a': 1, 'b': 2, 'c': 3, 'e': 2}


15. Write a Python program to check a dictionary is empty or not.

In [6]:
my_dict = {'a': 1, 'b': 2, 'c': 1, 'c': 3, 'e': 2}
if len(my_dict)==0:
    print("dictionary is empty")
else:
    print("dictionary is not empty")

dictionary is not empty


16. Write a Python program to combine two dictionary adding values for common keys. 
d1 = {'a': 100, 'b': 200, 'c':300}
d2 = {'a': 300, 'b': 200, 'd':400}
Sample output: Counter({'a': 400, 'b': 400, 'd': 400, 'c': 300})


In [14]:
d1={'a':100,'b':200,'c':300}
d2={'a':300,'b':200,'d':400}
counter={}
for i,j in d1.items():
    d1[i]=j
    for x,y in d2.items():
        d2[x]=y
        if i==x:
            j=j+y
    counter[i]=j
    counter[x]=y   
            
print(counter)   

{'a': 400, 'd': 400, 'b': 400, 'c': 300}


17. Write a Python program to print all unique values in a dictionary. 
Sample Data : [{"V":"S001"}, {"V": "S002"}, {"VI": "S001"}, {"VI": "S005"}, {"VII":"S005"}, {"V":"S009"},{"VIII":"S007"}]
Expected Output : Unique Values: {'S005', 'S002', 'S007', 'S001', 'S009'}


In [3]:
d1=[{"V":"S001"}, {"V": "S002"}, {"VI": "S001"}, {"VI": "S005"}, {"VII":"S005"}, {"V":"S009"},{"VIII":"S007"}]


In [7]:
unique=set()
for d in d1:
    
    for value in d.values():
       # print(value)
        unique.add(value)
print(unique)

{'S007', 'S001', 'S005', 'S009', 'S002'}


18. Write a Python program to find the highest 3 values of corresponding keys in a dictionary. 


In [51]:
my_dict = {'a': 10, 'c': 5, 'b': 20, 'd': 3, 'e': 15}
sort_value=sorted(my_dict.values(),reverse=True)
top_3_value=sort_value[:3]
#top_3_value
keys=[x for x,y in my_dict.items() if y in top_3_value]
for i in range(len(top_3_value)):
    print(f"{keys[i]}:{top_3_value[i]}")
    

a:20
b:15
e:10


19. Write a Python program to sort a list alphabetically in a dictionary.

In [7]:
d = {
    'fruits': ['apple', 'banana', 'orange'],
    'animals': ['dog', 'cat', 'elephant'],
    'colors': ['red', 'blue', 'green']
}

In [17]:
sort_dic={}
for i,j in d.items():
        if type(j) is list:
            sort_dic[i]=sorted(j)
        else:
            sort_dic[i]=j
for i,j in sort_dic.items():
    print(i,':',j)
    
#print(sort_dic)           

fruits : ['apple', 'banana', 'orange']
animals : ['cat', 'dog', 'elephant']
colors : ['blue', 'green', 'red']


20.  Write a Python program to get the top three items in a shop. 
Sample data: {'item1': 45.50, 'item2':35, 'item3': 41.30, 'item4':55, 'item5': 24}
Expected Output:
item4 55
item1 45.5
item3 41.3

In [59]:
d = {'item1': 45.50, 'item2': 35, 'item3': 41.30, 'item4': 55, 'item5': 24}

sorted_values = sorted(d.values(), reverse=True)

top_values = sorted_values[:3]

keys = [key for key, value in d.items() if value in top_values]

for i in range(len(top_values)):
    print(f"{keys[i]}: {top_values[i]}")



item1: 55
item3: 45.5
item4: 41.3


21. Write a Python program to get the key, value and item in a dictionary. 



In [62]:
d={1:'mango',2:'chiku',3:'kiwi'}
for i,j in d.items():
    print("key: ",i)
    print("value: ",j)
    print("item: ",(i,j))
    print("")

key:  1
value:  mango
item:  (1, 'mango')

key:  2
value:  chiku
item:  (2, 'chiku')

key:  3
value:  kiwi
item:  (3, 'kiwi')



22. Write a Python program to print a dictionary line by line. 


In [63]:
for i,j in d.items():
    print(i,":",j)

1 : mango
2 : chiku
3 : kiwi


23.Write a Python program to check multiple keys exists in a dictionary. 


In [70]:
my_dict = {'a': 1, 'b': 2, 'c': 3, 'd': 4}
key_check=['a','g','c']
for k in key_check:
    if k in my_dict:
        print(k,' is exist')
    else:
        print(k,' not exist')

a  is exist
g  not exist
c  is exist


24. Write a Python program to count number of items in a dictionary value that is a list

In [80]:
 
d={1:'a',2:['b','c'],3:['d','e',5],4:['f','g'],5:10}
count=0
for j in d.values():
    if isinstance(j,list): 
        print(j)                  ###isinstance() is check value is  list
        count+=len(j)
print("total item of list: ",count)

['b', 'c']
['d', 'e', 5]
['f', 'g']
total item of list:  7


25. Write a Python program to replace dictionary values with their average. 


In [92]:
d={'a':10,'b':20,'c':60,'d':40}
average=sum(d.values())/len(d.values())
for i,j in d.items():
    d[i]=average
print(d)
    
    

{'a': 32.5, 'b': 32.5, 'c': 32.5, 'd': 32.5}


26.Write a Python program to create a dictionary of keys x, y, and z where each key has as value a list from 11-20, 21-30, and 31-40 respectively. Access the fifth value of each key from the dictionary. 
{'x': [11, 12, 13, 14, 15, 16, 17, 18, 19],
'y': [21, 22, 23, 24, 25, 26, 27, 28, 29],
'z': [31, 32, 33, 34, 35, 36, 37, 38, 39]}
15
25
35
x has value [11, 12, 13, 14, 15, 16, 17, 18, 19]
y has value [21, 22, 23, 24, 25, 26, 27, 28, 29]
z has value [31, 32, 33, 34, 35, 36, 37, 38, 39]


In [105]:
dic={'x': [11, 12, 13, 14, 15, 16, 17, 18, 19],
'y': [21, 22, 23, 24, 25, 26, 27, 28, 29],
'z': [31, 32, 33, 34, 35, 36, 37, 38, 39]}
#for i in dic.items():
print("x has 5th value: ",dic['x'][4])
print("y has 5th value: ",dic['y'][4])
print("z has 5th value: ",dic['z'][4])

x has 5th value:  15
y has 5th value:  25
z has 5th value:  35


27. Write a Python program to convert more than one list to nested dictionary. 
Original strings:
['S001', 'S002', 'S003', 'S004']
['Adina Park', 'Leyton Marsh', 'Duncan Boyle', 'Saim Richards']
[85, 98, 89, 92]
Nested dictionary:
[{'S001': {'Adina Park': 85}}, {'S002': {'Leyton Marsh': 98}}, {'S003': {'Duncan Boyle': 89}}, {'S004': {'Saim Richards': 92}}]


In [108]:
keys = ['S001', 'S002', 'S003', 'S004']
names = ['Adina Park', 'Leyton Marsh', 'Duncan Boyle', 'Saim Richards']
scores = [85, 98, 89, 92]
nested=[]
for keys,names,scores in zip(keys,names,scores):
    nested.append({keys : {names:scores}})
print(nested)
    

[{'S001': {'Adina Park': 85}}, {'S002': {'Leyton Marsh': 98}}, {'S003': {'Duncan Boyle': 89}}, {'S004': {'Saim Richards': 92}}]


28. Write a Python program to create a dictionary grouping a sequence of key-value pairs into a dictionary of lists. 
Original list:
[('yellow', 1), ('blue', 2), ('yellow', 3), ('blue', 4), ('red', 1)]
Grouping a sequence of key-value pairs into a dictionary of lists:
{'yellow': [1, 3], 'blue': [2, 4], 'red': [1]}


In [2]:
original_list = [('yellow', 1), ('blue', 2), ('yellow', 3), ('blue', 4), ('red', 1)]

grouped_dict = {}

for key, value in original_list:
    if key in grouped_dict:
        grouped_dict[key].append(value) 
    else:
        grouped_dict[key] = [value]  
print(grouped_dict)

{'yellow': [1, 3], 'blue': [2, 4], 'red': [1]}


29.Write a Python program to filter the height and width of students, which are stored in a dictionary. 
Original Dictionary:
{'Cierra Vega': (6.2, 70), 'Alden Cantrell': (5.9, 65), 'Kierra Gentry': (6.0, 68), 'Pierre Cox': (5.8, 66)}
Height > 6ft and Weight> 70kg:
{'Cierra Vega': (6.2, 70)}


In [14]:
students = {
    'Cierra Vega': (6.2, 70),
    'Alden Cantrell': (5.9, 65),
    'Kierra Gentry': (6.0, 68),
    'Pierre Cox': (5.8, 66)
}


filtered_students = {name: (height, weight) for name, (height, weight) in students.items() if height > 6.0 and weight >=70}

print("Height > 6ft and Weight > 70kg:")
print(filtered_students)

Height > 6ft and Weight > 70kg:
{'Cierra Vega': (6.2, 70)}


In [22]:
#30 .  
original_list_float = [{'x': '10.12', 'y': '20.23', 'z': '30'}, {'p': '40.00', 'q': '50.19', 'r': '60.99'}]

converted_list_float = [{key: float(value) for key, value in dictionary.items()} for dictionary in original_list_float]


print(converted_list_float)



[{'x': 10.12, 'y': 20.23, 'z': 30.0}, {'p': 40.0, 'q': 50.19, 'r': 60.99}]


31. A Python Dictionary contains List as value. Write a Python program to clear the list values in the said dictionary. 
Original Dictionary:
{'C1': [10, 20, 30], 'C2': [20, 30, 40], 'C3': [12, 34]}
Clear the list values in the said dictionary:
{'C1': [], 'C2': [], 'C3': []}

In [31]:
dic1={'C1': [10, 20, 30], 'C2': [20, 30, 40], 'C3': [12, 34]}
for key,value in dic1.items():
    #if isinstance(value,list):
        dic1[key]=[]  #dic1[key].clear()
print(dic1)       

{'C1': [], 'C2': [], 'C3': []}


32. Write a Python program to find the length of a given dictionary values. 
Original Dictionary:
{1: 'red', 2: 'green', 3: 'black', 4: 'white', 5: 'black'}
Length of dictionary values:
{'red': 3, 'green': 5, 'black': 5, 'white': 5}
Original Dictionary:
{'1': 'Austin Little', '2': 'Natasha Howard', '3': 'Alfred Mullins', '4': 'Jamie Rowe'}
Length of dictionary values:
{'Austin Little': 13, 'Natasha Howard': 14, 'Alfred Mullins': 14, 'Jamie Rowe': 10}

In [3]:
dic1={1: 'red', 2: 'green', 3: 'black', 4: 'white', 5: 'black'}
#for value in dic1.values():
    #print(f'{value} :', len(value))
len={value: len(value) for value in dic1.values()}
print(len)

{'red': 3, 'green': 5, 'black': 5, 'white': 5}


33. Write a Python program to access dictionary key's element by index. 
Expected Output:
physics
math
chemistry


In [11]:
my_dict = {'subject1': 'physics', 'subject2': 'math', 'subject3': 'chemistry'}
value=list(my_dict.values())
print(value[0])
print(value[1])
print(value[2])


physics
math
chemistry


34.  Write a Python program to convert a given dictionary into a list of lists. 
Original Dictionary:
{1: 'red', 2: 'green', 3: 'black', 4: 'white', 5: 'black'}
Convert the said dictionary into a list of lists:
[[1, 'red'], [2, 'green'], [3, 'black'], [4, 'white'], [5, 'black']]

In [4]:
dic1={1: 'red', 2: 'green', 3: 'black', 4: 'white', 5: 'black'}
new_list=[[key,value] for key,value in dic1.items()]
print(new_list)

[[1, 'red'], [2, 'green'], [3, 'black'], [4, 'white'], [5, 'black']]


35. Write a Python program to filter even numbers from a given dictionary values. 
 Write a Python program to filter even numbers from a given dictionary values. 
Original Dictionary:
{'V': [1, 4, 6, 10], 'VI': [1, 4, 12], 'VII': [1, 3, 8]}
Filter even numbers from said dictionary values:
{'V': [4, 6, 10], 'VI': [4, 12], 'VII': [8]}


In [10]:
dic1 = {'V': [1, 4, 6, 10], 'VI': [1, 4, 12], 'VII': [1, 3, 8]}

new_dic = {i: [x for x in j if x%2==0] for i,j in dic1.items()}

print(new_dic)

{'V': [4, 6, 10], 'VI': [4, 12], 'VII': [8]}


36. Write a Python program to count the frequency in a given dictionary. 
Original Dictionary:
{'V': 10, 'VI': 10, 'VII': 40, 'VIII': 20, 'IX': 70, 'X': 80, 'XI': 40, 'XII': 20}
Count the frequency of the said dictionary:
Counter({10: 2, 40: 2, 20: 2, 70: 1, 80: 1})

In [30]:
dic1={'V': 10, 'VI': 10, 'VII': 40, 'VIII': 20, 'IX': 70, 'X': 80, 'XI': 40, 'XII': 20}
new_dic={}
for value in dic1.values():
   # print(value)
    if value in new_dic:
        new_dic[value]+=1
    else:
        new_dic[value]=1
print(new_dic)




{10: 2, 40: 2, 20: 2, 70: 1, 80: 1}


37. Write a Python program to combines two or more dictionaries, creating a list of values for each key. 
Sample Output:
Original dictionaries:
{'w': 50, 'x': 100, 'y': 'Green', 'z': 400}
{'x': 300, 'y': 'Red', 'z': 600}
Combined dictionaries, creating a list of values for each key:
{'w': [50], 'x': [100, 300], 'y': ['Green', 'Red'], 'z': [400, 600]}

In [23]:
dic1 = {'w': 50, 'x': 100, 'y': 'Green', 'z': 400}
dic2 = {'x': 300, 'y': 'Red', 'z': 600}
new_dic = {}

for d in (dic1, dic2):
    for key, value in d.items():
        if key in new_dic:
            new_dic[key].append(value)
        else:
            new_dic[key] = [value]

print(new_dic)

    

{'w': [50], 'x': [100, 300], 'y': ['Green', 'Red'], 'z': [400, 600]}


38. . Write a Python program to combine two lists into a dictionary, where the elements of the first one serve as the keys and the elements of the second one serve as the values. The values of the first list need to be unique and hashable. 
Sample Output:
Original lists:
['a', 'b', 'c', 'd', 'e', 'f']
[1, 2, 3, 4, 5]
Combine the values of the said two lists into a dictionary:
{'a': 1, 'b': 2, 'c': 3, 'd': 4, 'e': 5}

In [24]:
list1=['a', 'b', 'c', 'd', 'e', 'f']
list2=[1, 2, 3, 4, 5]
new_dic=dict(zip(list1,list2))
print(new_dic)

{'a': 1, 'b': 2, 'c': 3, 'd': 4, 'e': 5}


39. Write a Python program to convert given a dictionary to a list of tuples. 
Sample Output:
Original Dictionary:
{'Red': 1, 'Green': 3, 'White': 5, 'Black': 2, 'Pink': 4}
Convert the said dictionary to a list of tuples:
[('Red', 1), ('Green', 3), ('White', 5), ('Black', 2), ('Pink', 4)]

In [28]:
dic1={'Red': 1, 'Green': 3, 'White': 5, 'Black': 2, 'Pink': 4}

tpl=[(i,j) for i,j in dic1.items()]
tpl

[('Red', 1), ('Green', 3), ('White', 5), ('Black', 2), ('Pink', 4)]

40. Write a Python program to find the key of the maximum value in a dictionary. 
Sample Output:
Original dictionary elements:
{'Theodore': 19, 'Roxanne': 22, 'Mathew': 21, 'Betty': 20}
Finds the key of the maximum and minimum value of the said dictionary:
('Roxanne', 'Theodore')

In [43]:
dic1={'Theodore': 19, 'Roxanne': 22, 'Mathew': 21, 'Betty': 20}

max_val=max(dic1.values())
min_val=min(dic1.values())

for key,value in dic1.items():
    if max_val==value:
        print("maximum value of key:" ,key)
    if min_val == value:
        print("minimum value of key: ",key)

minimum value of key:  Theodore
maximum value of key: Roxanne


In [45]:
max_val=max(dic1,key=dic1.get)
min_val=min(dic1,key=dic1.get)
print((min_val,max_val))

('Theodore', 'Roxanne')
